tests/validate-response.test.js

```diff
@@ -103,6 +112,21 @@ describe('validateResponse', () => {
       schemas = (await generateSchemas(doc)).schemas;
     });
 
+    it('validates /books', () => {
```

Comment:
I do not understand how this test correlates with the edit in src/valid-response.js.

As I see, the bug was related to the fact that if a user passed an object with `statusCode` key into `validateResponse`, the function ignored the checks related to method or the number of URL segments of the request.

So, I expect that if we fix some bug, we add the tests that triggers this bug. But I can't understand how the added test is related to the bug 🤔 

I mean, when I revert the change from src/valid-response.js and run the tests, they work:

```
$ git diff
diff --git a/src/validate-response.js b/src/validate-response.js
index 42363a5..8e697ff 100644
--- a/src/validate-response.js
+++ b/src/validate-response.js
@@ -71,7 +71,7 @@ export function validateResponse({ method, statusCode, url, data, schemas, baseP
   ));

   if (statusCode) {
-    foundSchemas = foundSchemas.filter(schema => (schema.statusCode === statusCode));
+    foundSchemas = schemas.filter(schema => (schema.statusCode === statusCode));
   }

   if (statusField) {
```

```
$ npm test

> @funboxteam/api-validator@4.0.0 test /home/ai/funbox-oss/api-validator
> mocha --require @babel/register tests/*.test.js

  validateResponse
    ✔ handles arrays in parameters
    ✔ handles parameters without attributes
    ✔ accepts null value for nullable referenced attribute
    simple doc
      ✔ handles valid response
      
# ...and so on

    message with no content
      ✔ handles valid message (no data)
      ✔ handles invalid message (has some data)


  64 passing (146ms)
```

So, it looks like tests do not cover this bug. I guess we should add something like this:

```diff
diff --git a/tests/validate-response.test.js b/tests/validate-response.test.js
index bf792b1..d384253 100644
--- a/tests/validate-response.test.js
+++ b/tests/validate-response.test.js
@@ -880,6 +880,10 @@ describe('validateResponse', () => {
       const doc = `
 # My API
 
+# POST /example/post
++ Response 200 (text/plain)
+  Запрос выполнен успешно
+
 # GET /example
 
 + Response 200 (text/plain)
@@ -918,6 +922,16 @@ describe('validateResponse', () => {
       });
       assert.equal(result.status, validationStatus.schemaNotFound);
     });
+
+    it('handles URL check properly when status code is passed', () => {
+      const result = validateResponse({
+        method: 'POST',
+        statusCode: 200,
+        url: '/example',
+        schemas,
+      });
+      assert.equal(result.status, validationStatus.schemaNotFound);
+    });
   });
 });
```

Because this test fails without the changes of this PR:

```
$ npm test

> @funboxteam/api-validator@4.0.0 test /home/ai/funbox-oss/api-validator
> mocha --require @babel/register tests/*.test.js

  validateResponse
    ✔ handles arrays in parameters
    ✔ handles parameters without attributes
    ✔ accepts null value for nullable referenced attribute
    
# ...and so on

      AssertionError [ERR_ASSERTION]: 'valid' == 'schemaNotFound'
      + expected - actual

      -valid
      +schemaNotFound

      at Context.<anonymous> (tests/validate-response.test.js:933:14)

npm ERR! Test failed.  See above for more details.
```

It does not mean that the current change of the tests is useless. I just do not understand why they have been made. 

But we definitely should add a test that covers the bug.

Reply:
Btw, I've copied `Запрос выполнен успешно` from the file. It would be great if you translate this part to English. There are just two lines in the validate-response.test.js.

Reply:
You are right. My test was quite useless. So, I removed it and added a new one instead, as you recommended.